src/Giraffe/HttpHandlers.fs

```diff
@@ -387,23 +417,7 @@ let negotiateWith (negotiationRules    : IDictionary<string, obj -> HttpHandler>
 /// text/xml         -> serializes object to XML
 /// text/plain       -> returns object's ToString() result
 let negotiate (responseObj : obj) : HttpHandler =
-    negotiateWith
-        // Default negotiation rules
-        (dict [
-            "*/*"             , json
-            "application/json", json
-            "application/xml" , xml
-            "text/xml"        , xml
-            "text/plain"      , fun x -> x.ToString() |> text
-        ])
-        // Default unacceptable HttpHandler
-        (fun (next : HttpFunc) (ctx : HttpContext) ->
-            (setStatusCode 406
-            >=> ((ctx.Request.Headers.["Accept"]).ToString()
-                |> sprintf "%s is unacceptable by the server."
-                |> text)) next ctx)
-        // response object
-        responseObj
+    negotiateWith NegotiationRules UnacceptableHandler responseObj
```

Comment:
Are we able to use ETA reduction here such that 
`let negotiate = negotiateWith NegotiationRules UnacceptableHandler`
so the function is pre-applied and doesn't need to wrap closure each time at runtime?

Sorry very minor tweak but we try to pre-build as much as possible to leave minimal to runtime.

Reply:
that is compiler decision if you ask me.